packages/client/src/runtime/RequestHandler.ts

```diff
@@ -77,41 +71,62 @@ type ApplyExtensionsParams = {
 
 export class RequestHandler {
   client: Client
-  dataloader: DataLoader<Request>
+  dataloader: DataLoader<RequestParams>
   private logEmitter?: EventEmitter
 
   constructor(client: Client, logEmitter?: EventEmitter) {
     this.logEmitter = logEmitter
     this.client = client
     this.dataloader = new DataLoader({
       batchLoader: (requests) => {
-        const { transaction, protocolEncoder, otelParentCtx } = requests[0]
-        const queries = protocolEncoder.createBatch(requests.map((r) => r.protocolMessage))
-        const traceparent = this.client._tracingHelper.getTraceParent(otelParentCtx)
-
-        // TODO: pass the child information to QE for it to issue links to queries
-        // const links = requests.map((r) => trace.getSpanContext(r.otelChildCtx!))
-
-        const containsWrite = requests.some((r) => r.protocolMessage.isWrite())
-
-        return this.client._engine.requestBatch(queries, {
-          traceparent,
-          transaction: getTransactionOptions(transaction),
-          containsWrite,
-          customDataProxyFetch: requests[0].customDataProxyFetch,
-        })
+        return applyQueryBatchExtensions(
```

Comment:
Could we make this function a partial application so we don't need to pass request again?